lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,223 @@
+    })
+  }
+
+  async fetchAllPages(endpoint, params, data) {
+    const fetchAllPagesAccumular = (page, dataAccumular = []) =>
+      this.fetch(endpoint, { ...params, page }, data, true).then(response => {
+        const responseData = Array.isArray(response.data.data)
+          ? response.data.data
+          : [response.data.data]
+        const result = [...responseData, ...dataAccumular]
+
+        if (
+          response.data.meta.pagination.totalPages >
+          response.data.meta.pagination.page
+        ) {
+          return fetchAllPagesAccumular(
+            response.data.meta.pagination.page + 1,
+            result
+          )
+        }
+
+        return result
+      })
+
+    return fetchAllPagesAccumular(params.page || 1)
+  }
+
+  async walletBalance(walletId, index, minConfirmations = 1) {
+    return this.fetchAllPages('transactions', {
+      wallet_id: walletId,
+      per_page: CardanoAPI.PER_PAGE_MAXIMUM,
+      account_index: index // TODO: ues this or not? we will count all deposit on wallet or only for one index?
+    })
+      .then(transactions =>
+        transactions
+          .map(item => {
+            const sign = { incoming: +1, outgoing: -1 }[item.direction]
+            return {
+              isConfirmed: item.confirmations >= minConfirmations,
+              amount: sign * item.amount
+            }
+          })
+          .reduce(
+            (acc, item) => {
+              acc.confirmed += item.isConfirmed ? item.amount : 0
+              acc.total += item.amount
+              return acc
+            },
+            {
+              confirmed: 0,
+              total: 0
+            }
+          )
+      )
+      .then(({ confirmed, total }) => ({
+        confirmed: BN(confirmed).round(),
+        total: BN(total).round()
+      }))
+  }
+
+  sumAmountByAddress(movements, address) {
+    return movements
+      .filter(movement => movement.address === address)
+      .map(movement => movement.amount)
+      .reduce((acc, amount) => acc + amount, 0)
+  }
+
+  async addressBalance(
+    address,
+    minConfirmations,
+    mode = CardanoAPI.INCLUDE_ALL_TRANSACTIONS
+  ) {
+    return this.fetchAllPages('transactions', {
+      address: address,
+      per_page: CardanoAPI.PER_PAGE_MAXIMUM
+    })
+      .then(transactions =>
+        transactions
+          .map(transaction => {
+            let amount = 0
+
+            if (mode & CardanoAPI.INCLUDE_INCOME_TRANSACTIONS) {
+              amount -= this.sumAmountByAddress(transaction.inputs, address)
+            }
+
+            if (mode & CardanoAPI.INCLUDE_OUTCOME_TRANSACTIONS) {
+              amount += this.sumAmountByAddress(transaction.output, address)
+            }
+
+            return {
+              isConfirmed: transaction.confirmations >= minConfirmations,
+              amount
+            }
+          })
+          .reduce(
+            (acc, item) => {
+              acc.confirmed += item.isConfirmed ? item.amount : 0
```

Comment:
please, be more functional and do not modify object. Instead, just return new one